addons/mail/static/src/components/message/message.js

```diff
@@ -274,24 +275,66 @@ class Message extends Component {
+                    break;
+                /**
+                 * many2one formatter exists but is expecting id/name_get or data
+                 * object but only the target record name is known in this context.
+                 *
+                 * Selection formatter exists but requires knowing all
+                 * possibilities and they are not given in this context.
+                 */
+                case 'char':
+                case 'many2one':
+                case 'selection':
+                    value.old_value = format.char(value.old_value, undefined, { forceString: true });
+                    value.new_value = format.char(value.new_value, undefined, { forceString: true });
+                    break;
+                case 'date':
+                    if (value.old_value) {
+                        value.old_value = moment.utc(value.old_value);
+                    }
+                    if (value.new_value) {
+                        value.new_value = moment.utc(value.new_value);
+                    }
+                    value.old_value = format.date(value.old_value, undefined, { forceString: true });
+                    value.new_value = format.date(value.new_value, undefined, { forceString: true });
+                    break;
+                case 'datetime':
+                    if (value.old_value) {
+                        value.old_value = moment.utc(value.old_value);
+                    }
+                    if (value.new_value) {
+                        value.new_value = moment.utc(value.new_value);
+                    }
+                    value.old_value = format.datetime(value.old_value, undefined, { forceString: true });
+                    value.new_value = format.datetime(value.new_value, undefined, { forceString: true });
+                    break;
+                case 'float':
+                    value.old_value = format.float(value.old_value, undefined, { forceString: true });
+                    value.new_value = format.float(value.new_value, undefined, { forceString: true });
+                    break;
+                case 'integer':
+                    value.old_value = format.integer(value.old_value, undefined, { forceString: true });
+                    value.new_value = format.integer(value.new_value, undefined, { forceString: true });
+                    break;
+                case 'monetary':
+                    value.old_value = format.monetary(value.old_value, undefined, { forceString: true });
+                    value.new_value = format.monetary(value.new_value, undefined, { forceString: true });
+                    break;
+                case 'text':
+                    value.old_value = format.text(value.old_value, undefined, { forceString: true });
+                    value.new_value = format.text(value.new_value, undefined, { forceString: true });
+                    break;
```

Comment:
No `forceString` neither

Reply:
Yeah in summary only boolean and monetary have it, I'll remove it from others.